week08_pomdp/practice_pytorch.ipynb

```diff
@@ -591,25 +661,110 @@
+    "      plt.plot(\n",
+    "          moving_average(np.array(entropy_history), span=100),\n",
+    "          label=\"entropy ewma@100\",\n",
+    "      )\n",
+    "      plt.grid()\n",
+    "      plt.legend()\n",
+    "\n",
+    "      plt.subplot(3, 3, 3)\n",
+    "      plt.title(\"Entropy Loss (smoothened)\")\n",
+    "      plt.plot(\n",
+    "          moving_average(np.array(entropy_loss_history), span=100),\n",
+    "          label=\"entropy-loss ewma@100\",\n",
+    "      )\n",
+    "      plt.grid()\n",
+    "      plt.legend()\n",
+    "\n",
+    "      plt.subplot(3, 3, 4)\n",
+    "      plt.title(\"loss (smoothened)\")\n",
+    "      plt.plot(np.array(loss_history), label=\"loss raw\")\n",
+    "      plt.plot(moving_average(np.array(loss_history), span=10), label=\"loss ewma@10\")\n",
+    "      plt.grid()\n",
+    "      plt.legend()\n",
+    "\n",
+    "      plt.subplot(3, 3, 5)\n",
+    "      plt.title(\"Value Loss (smoothened)\")\n",
+    "      plt.plot(np.array(v_loss), label=\"V-loss raw\")\n",
+    "      plt.plot(moving_average(np.array(v_loss), span=10), label=\"V - loss ewma@10\")\n",
+    "      plt.grid()\n",
+    "      plt.legend()\n",
+    "\n",
+    "      plt.subplot(3, 3, 6)\n",
+    "      plt.title(\"Policy Loss (smoothened)\")\n",
+    "      plt.plot(np.array(Jhat), label=\"J-loss raw\")\n",
+    "      plt.plot(moving_average(np.array(Jhat), span=10), label=\"J - loss ewma@10\")\n",
+    "      plt.grid()\n",
+    "      plt.legend()\n",
+    "\n",
+    "      plt.subplot(3, 3, 7)\n",
+    "      plt.title(\"Grad norm history (smoothened)\")\n",
+    "      plt.plot(\n",
+    "          moving_average(np.array(grad_norm_history), span=100),\n",
+    "          label=\"grad norm ewma@100\",\n",
+    "      )\n",
+    "      plt.grid()\n",
+    "      plt.legend()\n",
+    "      plt.tight_layout()\n",
+    "      plt.show()\n",
+    "      \n",
+    "      if rewards_history[-1] > 10000:\n",
+    "        print(\"\\nYour agent has just beaten the homework record by: \", rewards_history[-1])\n",
+    "        break\n",
+    "      elif rewards_history[-1] >= 10000:\n",
+    "        print(\"\\nYour agent has just passed the minimum homework threshold with score: \", rewards_history[-1])\n",
+    "        break\n",
+    "      elif 8000 <= rewards_history[-1] <= 8499:\n",
+    "        print(\"\\nYour agent has just get 'Red' Belt with score: \", rewards_history[-1])\n",
+    "      elif 8500 <= rewards_history[-1] <= 9999:\n",
+    "        print(\"\\nYour agent has just get 'Red-Black' Belt with score: \", rewards_history[-1])\n",
+    "\n",
+    "      elif rewards_history[-1] >= 500:\n",
+    "        print(\"\\nYour agent has earned the yellow belt: \", rewards_history[-1])\n",
+    "    \n",
+    "  if i % 500 == 0:\n",
+    "    rollout_len = min(40, rollout_len + 1)\n",
+    "    print(f\"\\nNumber of Interactions per steps: {rollout_len}\")"
```

Comment:
It's worth pointing out explicitly (in a Markdown text block) that there is this rollout length schedule and the rationale for it.

Reply:
I think you guys have already done that!? And that's why I added it!!

![image](https://user-images.githubusercontent.com/72295771/205725120-0270a835-f19a-4579-9e9b-06b6756dc3fa.png)